Not needed to build llvm, but required for other applications that
include its header files, such as rust when built with external llvm.


--- llvm-13.0.1/llvm/include/llvm/IR/LLVMContext.h~	2022-01-20 21:31:59.000000000 +0000
+++ llvm-13.0.1/llvm/include/llvm/IR/LLVMContext.h	2023-04-02 19:08:39.158915907 +0000
@@ -14,6 +14,14 @@
 #ifndef LLVM_IR_LLVMCONTEXT_H
 #define LLVM_IR_LLVMCONTEXT_H
 
+// Avoid collision in llvm header files with CS and FS as these are
+// defined in Solaris regset.h header (intel only issue + built with gcc).
+#if defined(__sun__) && !defined(__sparc__)
+#include <sys/regset.h>
+#undef CS
+#undef FS
+#endif
+
 #include "llvm-c/Types.h"
 #include "llvm/IR/DiagnosticHandler.h"
 #include "llvm/Support/CBindingWrapping.h"
--- llvm-16.0.6/llvm/include/llvm/Object/ObjectFile.h.~1~	2023-06-19 18:18:25.796349597 +0000
+++ llvm-16.0.6/llvm/include/llvm/Object/ObjectFile.h	2023-06-19 18:19:22.996948761 +0000
@@ -13,6 +13,14 @@
 #ifndef LLVM_OBJECT_OBJECTFILE_H
 #define LLVM_OBJECT_OBJECTFILE_H
 
+// Avoid collision in llvm header files with CS and FS as these are
+// defined in Solaris regset.h header (intel only issue + built with gcc).
+#if defined(__sun__) && !defined(__sparc__)
+#include <sys/regset.h>
+#undef CS
+#undef FS
+#endif
+
 #include "llvm/ADT/ArrayRef.h"
 #include "llvm/ADT/Hashing.h"
 #include "llvm/ADT/StringRef.h"
--- llvm-16.0.6/llvm/include/llvm/ADT/SmallVector.h.~1~	2023-06-19 18:18:25.797393950 +0000
+++ llvm-16.0.6/llvm/include/llvm/ADT/SmallVector.h	2023-06-19 18:19:52.352646921 +0000
@@ -14,6 +14,14 @@
 #ifndef LLVM_ADT_SMALLVECTOR_H
 #define LLVM_ADT_SMALLVECTOR_H
 
+// Avoid collision in llvm header files with CS and FS as these are
+// defined in Solaris regset.h header (intel only issue + built with gcc).
+#if defined(__sun__) && !defined(__sparc__)
+#include <sys/regset.h>
+#undef CS
+#undef FS
+#endif
+
 #include "llvm/Support/Compiler.h"
 #include "llvm/Support/type_traits.h"
 #include <algorithm>
--- llvm-13.0.1/llvm/include/llvm/Linker/Linker.h~	2022-01-20 21:31:59.000000000 +0000
+++ llvm-13.0.1/llvm/include/llvm/Linker/Linker.h	2023-04-02 19:19:02.125035530 +0000
@@ -9,6 +9,14 @@
 #ifndef LLVM_LINKER_LINKER_H
 #define LLVM_LINKER_LINKER_H
 
+// Avoid collision in llvm header files with CS and FS as these are
+// defined in Solaris regset.h header (intel only issue + built with gcc).
+#if defined(__sun__) && !defined(__sparc__)
+#include <sys/regset.h>
+#undef CS
+#undef FS
+#endif
+
 #include "llvm/ADT/StringSet.h"
 #include "llvm/Linker/IRMover.h"
 
--- llvm-13.0.1/llvm/include/llvm/IR/DiagnosticInfo.h~	2022-01-20 21:31:59.000000000 +0000
+++ llvm-13.0.1/llvm/include/llvm/IR/DiagnosticInfo.h	2023-04-02 19:20:23.408879574 +0000
@@ -14,6 +14,14 @@
 #ifndef LLVM_IR_DIAGNOSTICINFO_H
 #define LLVM_IR_DIAGNOSTICINFO_H
 
+// Avoid collision in llvm header files with CS and FS as these are
+// defined in Solaris regset.h header (intel only issue + built with gcc).
+#if defined(__sun__) && !defined(__sparc__)
+#include <sys/regset.h>
+#undef CS
+#undef FS
+#endif
+
 #include "llvm-c/Types.h"
 #include "llvm/ADT/Optional.h"
 #include "llvm/ADT/SmallVector.h"
